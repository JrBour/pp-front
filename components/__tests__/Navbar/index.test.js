import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import Navbar from '@/components/Navbar'

const localVue = createLocalVue();
localVue.use(Vuex)

describe('Navbar', () => {
  let wrapper
  let store
  let state

  beforeEach(() => {
    state = {
      notifications: [
        {
          isRead: false
        }
      ]
    }
    store = new Vuex.Store({ state })
    wrapper = shallowMount(Navbar, {
      store,
      localVue,
      mocks: {
          $router: {
            push: jest.fn(),
            history: {
              current: {
                path: 'cc'
              }
            }
          }
      },
    })
  })

  describe('computed', () => {
    it('should return 0 when all notification are read', () => {
      store = new Vuex.Store({ })
      wrapper = shallowMount(Navbar, {
        store,
        localVue,
        mocks: {
            $router: {
              push: jest.fn(),
              history: {
                current: {
                  path: 'cc'
                }
              }
            }
        },
      })
      expect(wrapper.vm.notifications).toBe(0)
    })
  })

  describe('snapshot', () => {
    it('renders properly', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
