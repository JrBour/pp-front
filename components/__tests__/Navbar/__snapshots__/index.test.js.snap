// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar snapshot renders properly 1`] = `
<nav class="navbar">
  <ul class="navbar__wrapper">
    <li><img src="" alt="home"></li>
    <li><img src="" alt="album"></li>
    <li><img src="" alt="camera" class="navbar__item_camera"></li>
    <li>
      <div class="notification__item"><img src="" alt="notifications" class="navbar__item_bell">
        <div class="notifications__number">
          <p>1</p>
        </div>
      </div>
    </li>
    <li><img src="" alt="profile"></li>
  </ul>
</nav>
`;
